src/aws-cpp-sdk-core/include/smithy/identity/signer/built-in/SigV4Signer.h

```diff
@@ -61,4 +61,51 @@ namespace smithy {
         Aws::String m_region;
         Aws::Client::AWSAuthV4Signer legacySigner;
     };
+
+    class AWSAuthEventStreamV4Signer : public AwsSignerBase<AwsCredentialIdentityBase> {
```

Comment:
place it in a separate file, so that 400+ service clients not-using-eventstreams won't have to compile this code.